olive/cache.py

```diff
@@ -465,20 +455,24 @@ def cache_run(self, model_id: str, run_json_path: Path) -> None:
             logger.debug("Cached run %s to shared cache.", model_id)
         except Exception:
             logger.exception("Failed to cache run to shared cache.")
+            # delete all uploaded files if any upload fails
+            try:
+                retry_func(self.container_client_factory.delete_blob, [model_id])
+            except Exception:
+                logger.exception(
+                    "Upload model to shared cache failed. There might be some dirty files in the shared cache."
+                    "Please manually clean up. %s",
+                    model_id,
+                )
 
     def load_run(self, model_id: str, run_json_path: Path) -> Optional[Dict]:
         blob = f"{model_id}/run.json"
         try:
             if not self.exist_in_shared_cache(blob):
                 logger.info("Run %s is not found in shared cache.", model_id)
                 return {}
-            blob_client = self.container_client.get_blob_client(blob)
             logger.info("Downloading %s to %s", blob, run_json_path)
-
-            with open(run_json_path, "wb") as download_file:
-                download_stream = blob_client.download_blob()
-                download_file.write(download_stream.readall())
-
+            retry_func(self.container_client_factory.download_blob, [blob, run_json_path])
```

Comment:
would it be possible to move the retry part into the container factory method implementation itself? I see that all method call here are wrapped in retry_func.

Reply:
Sounds good to me. Done.